### PR TITLE
handle num_samples=-1

### PR DESCRIPTION
**Pull request by sonichi**
## Why are these changes needed?

Fix a bug when `num_samples=-1` is passed to `BlendSearch.set_search_properties()`.

## Checks

- [x] I've used [pre-commit](https://microsoft.github.io/FLAML/docs/Contribute#pre-commit) to lint the changes in this PR, or I've made sure [lint with flake8](https://github.com/microsoft/FLAML/blob/816a82a1155b4de4705b21a615ccdff67c6da379/.github/workflows/python-package.yml#L54-L59) output is two 0s.
- [ ] I've included any doc changes needed for https://microsoft.github.io/FLAML/. See https://microsoft.github.io/FLAML/docs/Contribute#documentation to build and test documentation locally.
- [ ] I've added tests (if relevant) corresponding to the changes introduced in this PR.
- [x] I've made sure all auto checks have passed.

### CODE REVIEW

File: flaml/tune/searcher/blendsearch.py [hunk 1]
```diff
@@ -314,11 +314,12 @@ def set_search_properties(
...
                 self._metric_target = spec.get("metric_target")
-            if "num_samples" in spec:
+            num_samples = spec.get("num_samples")
+            if num_samples is not None:
                 self._num_samples = (
-                    spec["num_samples"]
-                    + len(self._result)
-                    + len(self._trial_proposed_by)
+                    (num_samples + len(self._result) + len(self._trial_proposed_by))
+                    if num_samples > 0
```
Comment (ZviBaratz) [inline flaml/tune/searcher/blendsearch.py:321]: Just to make sure – what is the expected behavior for `num_samples=0`? Does this cover that case appropriately?

**Comment by thinkall** [comment]
Some tests failed:

https://github.com/microsoft/FLAML/actions/runs/3871415089/jobs/6599189226#step:12:1338

```
FAILED test/automl/test_warmstart.py::TestWarmStart::test_FLAML_sample_size_in_starting_points - ray.exceptions.RuntimeEnvSetupError: Failed to setup runtime environment.
Could not create the actor because its associated runtime env failed to be created.
Failed to create runtime environment {"envVars": {"TUNE_ORIG_WORKING_DIR": "/Users/runner/work/FLAML/FLAML"}} because the Ray agent couldn't be started due to the port conflict. See `dashboard_agent.log` for more details. To solve the problem, start Ray with a hard-coded agent port. `ray start --dashboard-agent-grpc-port [port]` and make sure the port is not used by other processes.
```

File: flaml/tune/searcher/blendsearch.py (hunk 1 above)
Comment (sonichi) [reply flaml/tune/searcher/blendsearch.py]: `num_samples=0` is treated the same as `num_samples=-1`. This behaviour has not been documented. So I added a comment.

**Comment by sonichi** [comment]
> Some tests failed:
> 
> https://github.com/microsoft/FLAML/actions/runs/3871415089/jobs/6599189226#step:12:1338
> 
> ```
> FAILED test/automl/test_warmstart.py::TestWarmStart::test_FLAML_sample_size_in_starting_points - ray.exceptions.RuntimeEnvSetupError: Failed to setup runtime environment.
> Could not create the actor because its associated runtime env failed to be created.
> Failed to create runtime environment {"envVars": {"TUNE_ORIG_WORKING_DIR": "/Users/runner/work/FLAML/FLAML"}} because the Ray agent couldn't be started due to the port conflict. See `dashboard_agent.log` for more details. To solve the problem, start Ray with a hard-coded agent port. `ray start --dashboard-agent-grpc-port [port]` and make sure the port is not used by other processes.
> ```

All tests passed now.

**Review by thinkall — APPROVED** [review]
LGTM. Thanks!

**Review by ZviBaratz — APPROVED** [review]
👍👍

**Pull request merged**